src/synthesized_insight/check.py

```diff
@@ -109,6 +109,9 @@ def categorical(self, sr: pd.Series) -> bool:
         """Checks if the given series is categorical"""
         if pd.api.types.is_categorical_dtype(sr) is True:
             return True
+        sr = self.infer_dtype(sr)
+        if sr.dtype.kind == "M":  # TODO: Need to implement ability to deal with dates well in metrics
```

Comment:
Dates don't quite work yet for a lot of categorical metrics so we just return false here

Reply:
Agreed